src/gui/SampleWaveform.cpp

```diff
@@ -53,7 +54,7 @@ void SampleWaveform::visualize(Parameters parameters, QPainter& painter, const Q
 
 	int pixelIndex = 0;
 
-	for (int i = 0; i < maxFrames; i += resolution)
+	for (int i = 0; i < maxFrames; i += resolutionInt)
```

Comment:
Would it make sense to cast it directly inside the loop argument instead of introducing a variable?

Reply:
I wasn't sure whether the cast would be done over and over again or optimized into a one-time conversion, so I decided to make it a variable outside.

Reply:
It will be done only once from what I know. Would anyone confirm this?

Reply:
Compared the 2 ways and the performance difference is hardly noticeable. I'll go ahead and move the cast into the loop

Reply:
> It will be done only once from what I know. Would anyone confirm this?

Not sure but I think that depends on the level of optimization. I don't think it matters that much since it's not executing unless you poke the gui and resize something that forces a recompute of the wave.